frontend/packages/core/src/stepper.tsx

```diff
@@ -206,18 +218,31 @@ const Stepper = ({ activeStep, orientation = "horizontal", children }: StepperPr
           index: idx + 1,
           variant: "pending" as StepIconVariant,
         };
-        if (idx === activeStep) {
+
+        if (completed) {
+          if (completed[idx]) {
+            stepProps.variant = "success";
+          } else if (idx === activeStep) {
+            stepProps.variant = step.props.error ? "failed" : "active";
+          }
+        } else if (idx === activeStep) {
           stepProps.variant = step.props.error ? "failed" : "active";
         } else if (idx < activeStep) {
           stepProps.variant = "success";
         }
 
+        const label = step.props.label ?? `Step ${idx + 1}`;
+        const icon = <StepIcon {...stepProps} />;
+        const StepProps = completed ? { completed: completed[idx] } : {};
         return (
-          <MuiStep key={step.props.label}>
-            {/* eslint-disable-next-line react/no-unstable-nested-components */}
-            <MuiStepLabel StepIconComponent={() => <StepIcon {...stepProps} />}>
-              {step.props.label ?? `Step ${idx + 1}`}
-            </MuiStepLabel>
+          <MuiStep key={step.props.label} {...StepProps}>
+            {nonLinear ? (
+              <MuiStepButton onClick={() => handleStepClick(idx)} icon={icon}>
```

Comment:
We'll probably need a setting or something to change the color of the steps because right now it just all looks disabled and I don't think many would realize the other steps are clickable.

Reply:
I just updated the PR description with the new stepper look, let me know what you think.

Reply:
Hmm, what if we did something like this (as it currently is today).

So this would be the normal mode. In the non-linear mode it would have the background and number as the same color but the blue outline would not be there unless it's the selected step. What do you think?

Also, would you mind putting up some screenshots of normal / non-linear mode just so it's easy to verify the current state isn't changed.

![Screenshot 2024-07-15 at 11 27 29 AM](https://github.com/user-attachments/assets/56e6f752-20c9-417a-a51f-63506cda328e)

Reply:
Updated.

I decided to keep a light gray in the border, removing the border completely gives a sensation of not clickable element or that the styles are broken.